examples/kokkos/view/example_restrict.py

```diff
@@ -0,0 +1,522 @@
+
+   .. note::
+
+        According to :cite:label:`nvidia-cuda-ldg`, :code:`__ldg` (or any other cache strategy)
+        *supports all C++ fundamental types, CUDA vector types (except x3 components), and extended floating-point types*.
+        However, as of CUDA 13.1, it seems to only support up to :code:`double2`, *i.e.* up to
+        128-bit size objects. This seems contradictory with
+        :py:meth:`reprospect.test.features.Memory.max_transaction_size` that states 256-bit size objects
+        are supported as of :py:attr:`reprospect.tools.architecture.NVIDIAFamily.BLACKWELL` and CUDA 13.
+
+``.CONSTANT`` load
+------------------
+
+To ensure that a global load with ``.CONSTANT`` is used, use either of:
+
+#. :py:attr:`Method.GLOBAL_KERNEL`
+#. :py:attr:`Method.LDG_ACCESSOR`
+
+Strategies that fail preventing pointer aliasing
+------------------------------------------------
+
+The following strategies cannot be used to elude pointer aliasing:
+
+#. :py:attr:`Method.RESTRICT_ACCESSOR`
+
+   Writing a custom accessor, *e.g.* for `std::mdspan`_,
+   does not work.
+
+   This is the strategy employed in `PR 8151 <https://github.com/kokkos/kokkos/pull/8151>`_.
+
+#. :py:attr:`Method.RESTRICT_VIEW`
+
+   The implementation is similar to :py:attr:`Method.RESTRICT_ACCESSOR`.
+
+#. :py:attr:`Method.RESTRICT_MEMBER`
+
+   This strategy is also ineffective, as already noted in:
+
+   - https://discuss.pytorch.org/t/usage-of-restrict-in-cuda/150395
+   - https://forums.developer.nvidia.com/t/restrict-seems-to-be-ignored-for-base-pointers-in-structs-having-base-pointers-with-restrict-as-kernel-arguments-directly-works-as-expected/154020
+"""
+
+import logging
+import pathlib
+import re
+import sys
+import typing
+
+import pytest
+
+from reprospect.test import CMakeAwareTestCase, environment
+from reprospect.test.sass.composite import (
+    any_of,
+    instruction_is,
+    instructions_contain,
+    interleaved_instructions_are,
+)
+from reprospect.test.sass.controlflow.block import BasicBlockMatcher
+from reprospect.test.sass.instruction import (
+    LoadGlobalMatcher,
+    OpcodeModsWithOperandsMatcher,
+    StoreGlobalMatcher,
+)
+from reprospect.test.sass.instruction.integer import (
+    IntAdd3Matcher,
+    LEAMatcher,
+)
+from reprospect.test.sass.instruction.register import Register
+from reprospect.test.sass.matchers.add_int32 import AddInt32Matcher
+from reprospect.tools import ncu
+from reprospect.tools.binaries.cuobjdump import CuObjDump
+from reprospect.tools.sass.controlflow import ControlFlow, Graph
+from reprospect.tools.sass.decode import Decoder
+from reprospect.utils import detect
+
+if sys.version_info >= (3, 12):
+    from typing import override
+else:
+    from typing_extensions import override
+
+if sys.version_info >= (3, 11):
+    from enum import StrEnum
+else:
+    from backports.strenum.strenum import StrEnum
+
+
+class Method(StrEnum):
+    GLOBAL_KERNEL = 'global_kernel'
```

Comment:
We could think of writing having a list with comments once. Right now, there is a fist list with references, then a list with comments on the success/failures of the strategies, and here this list with more details about each method.